### ISSUE TITLE
Draggable sidenav

### ISSUE THREAD

**Issue by tam3r** [angular/material#1591]
Is it possible to make sidenav draggable? Not only google apps like gmail and play music demonstrate the approach, but also MD bottom sheets.

![bottomsheet](https://cloud.githubusercontent.com/assets/10559316/6307722/44270cbc-b94f-11e4-871b-78c443853a0f.gif)

It feels more responsive that way and would be an appreciated enhancement.

**Comment by andreabadesso** [comment]
+1

**Comment by ThomasBurleson** [comment]
@tam3r  - what about the other UX questions.
* Bottomsheet - How high should the bottomsheet be draggable? Why? What about snap-to when released?
* SideNav - Swipable I understand; why do you want this draggable?

**Comment by tam3r** [comment]
Bottomsheet looks good as is (to me at least). Not sure anything needs to be changed in terms of how far it can be pulled.
"Draggable" may not be the best word to describe how it is in native apps. I meant that it may be a good idea to let the sidenav position follow a finger position. When the menu is closed, moving the finger from the left edge of a screen to the right side makes the sidenav appear and move along with the finger. Then if the finger is released before the 50% of the menu width is reached then the sidenav slides back out. Otherwise it opens to the full width. of It does not go any further than full width of sidenav though. 
To me, that kind of behavior feels more responsive and "native". As far as I remember there have been requests for "slide from the edge to open a sidenav" feature, so that's killing two birds with one stone.

**Comment by rdesimone** [comment]
I agree to @tam3r  - (most of the sidenavs in JS frameworks are not implemented native like). There is another issue: when opening the sidenav with a swipe actually, the swipe event has to be attached to the view (content). In the case the swipe event is needed for another task (e.g. view change, swipe events on list items), the swipe event might not be available. Maybe a small element overlaying the content at the left (or right) border could be used to attach a swipe/touch event and to initiate the opening of the sidenav.

**Comment by gustavohenke** [comment]
I've suggested a review in #1890, which for me looks very similar to this one.
So, :+1: for what has been proposed here.

I've been implementing native-like swipe gestures in my app and it's rather complex. The community would benefit a lot if this was implemented in the ngMaterial core.

**Comment by rdesimone** [comment]
@marcysutton maybe take also a look at the core-drawer-panel from the Polymer project https://www.polymer-project.org/0.5/docs/elements/core-drawer-panel.html

**Comment by ThomasBurleson** [comment]
@rdesimone, @gustavohenke - agreed this will be a great feature. The team likes it... we have many other features to deliver first; which is why this request is in the **backlog**.

**Comment by NickBolles** [comment]
This might be nice to include on the `md-swipe-*` directives. It might work a little differently and be another issue though.

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.

**Comment by williamoliveira** [comment]
So this feature was essentially done, it had a pull request and now has been killed because of a new version that will take months to be released?

**Comment by Bakirelived** [comment]
what is the status of this?

**Comment by DevVersion** [comment]
We are hoping to re-address this after #9362 is available.
> This makes sure that there are no regressions with content inside of a draggable sidenav.

**Comment by WillsB3** [comment]
I see this has been re-opened for a while now.

Is there anything I can do to help the process of getting the existing PR merged?

**Comment by Splaktar** [comment]
Closing as duplicate of https://github.com/angular/material/issues/35.
